File: GotadeOuroDefault.Web/Modules/Produto/TipodeProduto/TipodeProdutoRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace GotadeOuroDefault.Produto;

[ConnectionKey("Default"), Module("Produto"), TableName("TipoProduto")]
[DisplayName("Tipode Produto"), InstanceName("Tipode Produto")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class TipodeProdutoRow : Row<TipodeProdutoRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id Tipo Produto"), Column("idTipoProduto"), PrimaryKey, NotNull, IdProperty]
    public int? IdTipoProduto
    {
        get => fields.IdTipoProduto[this];
        set => fields.IdTipoProduto[this] = value;
    }

    [DisplayName("Descricao"), Column("descricao"), Size(4000), QuickSearch, NameProperty]
    public string Descricao
    {
        get => fields.Descricao[this];
        set => fields.Descricao[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field IdTipoProduto;
        public StringField Descricao;

    }
}

File: GotadeOuroDefault.Web/Modules/ServerTypes/Produto/TipodeProdutoRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TipodeProdutoRow {
    IdTipoProduto?: number;
    Descricao?: string;
}

export abstract class TipodeProdutoRow {
    static readonly idProperty = 'IdTipoProduto';
    static readonly nameProperty = 'Descricao';
    static readonly localTextPrefix = 'Produto.TipodeProduto';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TipodeProdutoRow>();
}